docs: Redo design document to prevent destruction of additions & prevent bloat (#6480)
cataclysmbnteam/Cataclysm-BN: doc/src/content/docs/en/dev/explanation/design_document.md

2. Weird/wonky mechanics should be either reworked or removed. If there is value in retaining them,
   they should be clearly communicated to the player through UI.

3. Everything should fill its own niche or have its own purpose, e.g. we have too many firearms
   and clutter benefits nobody. If these additions are just more bloat, mainline mod it.
   Examples are guns, books, food, drugs, morale items, furniture variants, clothes.
   Opt-out is preferred for additional bloat, and Opt-in for things that change game theme.

>    e.g. Magical Nights, Aftershock, CRIT. When in doubt, ask.

Reviewer: I think most of the original text here is still relevant, and most of the proposed change would fit as a 3.1 or example on top of the original text. Reasoning is that the original text already lays out that "a dozen guns that differ only in name and +1 to damage value" should be mod items.

Inclusion defaults (opt-in/out) looks like it'd be best as a dot item under (3), or its own section.

Author: I could see that, let me do some edits.

Author: Should be resolved with latest commit.

2. weirdness is a sign that things are unintuitive. Somethings benefit from not being obvious. But
   things that are too far from people's expectations break the suspension of disbelief and require
   people to remember more disconnected elements.
3. More items means more interactions, more depth, but finding what you want or understanding
   what items are good when we have 30 AK pattern rifles and 30 AR-15 platforms.
   Distributions are also cluttered badly from increased variants. This is bad for gameplay.
   A more believable world has 10,000 firearm variants, a more playable world has 10.
   If in doubt, Add your new items as a mod package.
4. Sorting through complexity in menus isn't fun. menus are only fun when there are meaningful
   choices or tradeoffs being made there. (I.E. what to equip, what to carry, what to loot first,
   etc.)
5. and 6. are kind of common sense applications of prioritizing gameplay.

### Removals and re-adding

> Destruction of additions is a contentious problem in the community. To clarify official stance:

Reviewer 2: ```suggestion
Removal of content is a contentious problem in the community. To clarify official stance:
```
Destruction of additions just reads awkwardly to me.

terrain nearby.(terrain speed slowdowns. damage terrain, chokepoints, etc.) other clusters of
enemies. How much benefit is there to a fight? (zombie drops + nearby loot locations that are now
safe)

## Bloat

Bloat happens because we all want to add things that we personally like. But we can't add everything.

The rule on bloat is to save it for opt-out/opt-in mods, or third party mods.

Bloat is when there is no meaningful reason to have something in the game, A gun is a gun.
Clothes are clothes. We don't want 30 variations of the same T-Shirt but colored differently inside
the base game. Now you can't find what you want.

There is a difference between bloat, and flavor. Flavor is adding a missing cooking recipe.
We've historically not represented much European/African/Asian cooking. Some recipes exist,
Such as lutefisk and haggis, but a lot of recipes are generic or common American fare.

Bloat is adding rechambered versions of firearms, example being Vektor variants.

> Another example would be adding 30 variations of the bible, or 30 novels.

Reviewer 2: ```suggestion
Another example would be adding 30 variations of the bible, or 30 novels. Put it in an expanded literature mod, or a religion mod in the case of religious books.
```

> We don't need 20 types of beer, it just makes it harder to parse the comestibles menu.

Reviewer 2: ```suggestion
We don't need 20 types of beer, it just makes it harder to parse the comestibles menu. Put it in a (craft) brewery mod.
```
I think it makes sense to emphasize the mod angle on each example of bloat and give an example of the desired alternative.

> We don't need clothes whose only distinction is a different player sprite. Mod it.

Reviewer 2: ```suggestion
We don't need clothes whose only distinction is a different player sprite. Put it in a clothing/fashion mod.
```
